### ISSUE TITLE
Getting heap memory issue for the count query.

### ISSUE THREAD

**Issue by Aniket-Lonare** [orientechnologies/orientdb#10021]
## Expected behavior  
Should return the single record with the count number.

## Actual behavior  
Return multiple records according to the limit provided. If a limit is not provided then gives heap memory issues.

## Steps to reproduce  
We have a class named "Objects" with three properties: ID, Name, and Searchable. Indexes have been created for both the "Name" and "Searchable" properties as shown below:
Index for "Name" property:
``CREATE INDEX Objects.Name ON Objects (Name) UNIQUE_HASH_INDEX ENGINE HASH_INDEX``
Index for "Searchable" property:
``CREATE INDEX Objects.Searchable ON Objects (Searchable) NOTUNIQUE_HASH_INDEX ENGINE HASH_INDEX``
![issue_image_5](https://github.com/orientechnologies/orientdb/assets/53806566/2d4ee53a-13a1-4ed4-9dc9-51d390d19a3e)
After that, we created some records in the database.
Running the below query we are getting count results in multiple records as shown in the image.

`SELECT count(*) from Objects where Searchable = true`
![image](https://github.com/orientechnologies/orientdb/assets/53806566/5365bf67-b636-4371-be05-92dd6ccbc618)

This is explain view of the same query.
![issue_image_2](https://github.com/orientechnologies/orientdb/assets/53806566/9ce2bbd5-23bd-434d-92bc-ea129174079b)
If we run the query using orientjs(Without passing the limit) then it give us heap memory issue as shown below.
```
java.lang.OutOfMemoryError: Java heap space
Dumping heap to java_pid1.hprof ...
Heap dump file created [3286284170 bytes in 26.841 secs]
2023-08-28 12:44:40:095 INFO  Storage 'plocal:/orientdb/databases/SampleDB' is opened under OrientDB distribution : 3.2.22 (build ${buildNumber}, branch 3.2.22-without-embedded-validation) [OEnterpriseLocalPaginatedStorage]
2023-08-28 12:48:38:244 SEVER Error during WAL writing [CASDiskWriteAheadLog$RecordsWriter]
java.lang.OutOfMemoryError: Java heap space
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.CASDiskWriteAheadLog$RecordsWriter.run(CASDiskWriteAheadLog.java:1655)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)   

 

2023-08-28 12:48:38:268 SEVER Exception in thread 'OrientDB Write Cache Flush Task' [OScheduledThreadPoolExecutorWithLogging]
java.lang.OutOfMemoryError: Java heap space

 

2023-08-28 12:48:38:268 SEVER Uncaught exception in thread OrientDB (/172.26.0.7:2424) <- BinaryClient (/172.26.0.9:46822) [OUncaughtExceptionHandler]
java.lang.OutOfMemoryError: Java heap space

 

2023-08-28 12:48:38:271 SEVER Exception in thread 'OrientDB WAL Flush Task' [OScheduledThreadPoolExecutorWithLogging]
java.lang.OutOfMemoryError: Java heap space
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.CASDiskWriteAheadLog$RecordsWriter.run(CASDiskWriteAheadLog.java:1655)        
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

If we run the query without * or Adding any other property like ID we are getting results as expected as shown below images.

![issue_image_3](https://github.com/orientechnologies/orientdb/assets/53806566/97e14e46-abc8-4616-9ac8-712ae13febea)
![issue_image_4](https://github.com/orientechnologies/orientdb/assets/53806566/f88e4c25-c7e5-4402-98f8-5c5a1f5562ae)
![issue_image_6](https://github.com/orientechnologies/orientdb/assets/53806566/76c59425-ef81-4478-b337-8a6fdb65bba9)

If we delete the index then we are getting results as expected.
![issue_image_7](https://github.com/orientechnologies/orientdb/assets/53806566/df9965c8-8f81-41b8-8329-db39c69dce16)
![issue_image_8](https://github.com/orientechnologies/orientdb/assets/53806566/6f2ac668-4a58-4640-8089-4075d5c672c3)

In this scenario is there an index creation issue?
Please let me know if any additional information is needed to investigate this issue. 

Thank you.

**Comment by tglman** [comment]
Hi,

I tried to reproduce this case without any success, also your explain looks a bit strange to me, OrientDB has multiple steps usually like the `CALCULATE PROJECTIONS` you see in the second explain, not sure how can you make that explain.

**Comment by aniket23111998** [comment]
We have forwarded the sample database that is triggering this problem to one of your email address. Please inform us if any additional information is needed for a thorough analysis of this issue.

**Comment by tglman** [comment]
Hi,

I could reproduce this with a unit test and fix it, will be released next hotfix.

Regards

**Comment by Aniket-Lonare** [comment]
Okay, Thank you

**Comment by tglman** [comment]
Hi,

The release 3.2.23 is out, will close this, please re-open if you still experience similar problem.

Regards

**Comment by Aniket-Lonare** [comment]
It's resolved in 3.2.23, Thank you
